sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/ConditionalAggregates.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions.aggregate
+
+import org.apache.spark.sql.catalyst.analysis.TypeCheckResult
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.types._
+
+abstract class ConditionalAggregate(predicate: Expression) extends UnevaluableAggregate {
```

Comment:
We don't have to make this `abstract` for now since there's only single case

Reply:
You mean, `ConditionalAggregate` is not necessary and integrate it with `CountIf`?

Reply:
Yea, we won't need that for now.

Reply:
Okay, Thanks :)

Reply:
I just modified.